cmd/soroban-cli/src/tx/builder/transaction.rs

```diff
@@ -0,0 +1,54 @@
+use crate::xdr::{self, Memo, SequenceNumber, TransactionExt};
+
+use super::{Error, MuxedAccount};
+
+pub trait TxExt {
+    fn new_tx(
+        source: impl Into<MuxedAccount>,
+        fee: u32,
+        seq_num: impl Into<SequenceNumber>,
+        operation: xdr::Operation,
+    ) -> xdr::Transaction;
+
+    fn add_operation(self, operation: xdr::Operation) -> Result<xdr::Transaction, Error>;
+
+    fn add_memo(self, memo: Memo) -> xdr::Transaction;
+
+    fn add_cond(self, cond: xdr::Preconditions) -> xdr::Transaction;
+}
+
+impl TxExt for xdr::Transaction {
+    fn new_tx(
+        source_account: impl Into<MuxedAccount>,
+        fee: u32,
+        seq_num: impl Into<SequenceNumber>,
+        operation: xdr::Operation,
+    ) -> xdr::Transaction {
+        xdr::Transaction {
+            source_account: source_account.into().into(),
+            fee,
+            seq_num: seq_num.into(),
+            cond: soroban_env_host::xdr::Preconditions::None,
+            memo: Memo::None,
+            // Safe because we know we have only one operation
+            operations: unsafe { vec![operation].try_into().unwrap_unchecked() },
```

Comment:
I don't think we need to optimize cli for performance that much, we can take a runtime check and simplify to
 ```suggestion
            operations: vec![operation].try_into().unwrap(),
```

Reply:
The `vec![...]` should be unnecessary, I believe you can pass an array:

 ```suggestion
            operations: [operation].try_into().unwrap(),
```